javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -803,12 +765,19 @@ private void addLroMethods(Operation operation, ClientMethod.Builder builder, Li
                 .onlyRequiredParameters(false)
```

Comment:
We are still keeping `LongRunningBeginAsync`(and sync) with maximum overload without `Context`? 

This seems to have 3 overloads?
1. Maximum without Context
2. Required with minimum
3. Maximum with Context

Reply:
Yes, vanilla always provides all 3 overloads (similar to pageable). They are more developer facing (as usually vanilla Client is generated in implementation package), and dev would have max flexibility to choose the API they wish to call (as the implementation is not exposed to user).

We then trim down in fluent and dpg.

Reply:
PS: vanilla usually not actually enable LRO as it need to be configured https://github.com/Azure/autorest.java#settings-for-minimal-data-plane-clients